core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -861,6 +861,14 @@ int Cuda::impl_is_initialized() {
 
 void Cuda::impl_initialize(InitializationSettings const &settings) {
   Impl::CudaInternal::singleton().initialize(Impl::get_gpu(settings));
+
+  // In order to support setting an atexit hook for Kokkos::finalize
```

Comment:
```suggestion
  // In order to support registering `Kokkos::finalize` to be called at program termination (via std::atexit)
```
(not formatted feel free to ignore)